kubeadm/v1.15.0/KubeClusterHelper.psm1

```diff
@@ -0,0 +1,1515 @@
+{
+    Write-Host "Downloading CNI binaries for $NetworkMode to $CniPath"
+    CreateDirectory $CniPath\config
+    DownloadFile -Url "https://github.com/$Global:GithubSDNRepository/raw/$Global:GithubSDNBranch/Kubernetes/flannel/$NetworkMode/cni/config/cni.conf" -Destination $CniPath\config\cni.conf
+    DownloadFile -Url  "https://github.com/$Global:GithubSDNRepository/raw/$Global:GithubSDNBranch/Kubernetes/flannel/l2bridge/cni/flannel.exe" -Destination $CniPath\flannel.exe
+    DownloadFile -Url  "https://github.com/$Global:GithubSDNRepository/raw/$Global:GithubSDNBranch/Kubernetes/flannel/l2bridge/cni/host-local.exe" -Destination $CniPath\host-local.exe
+
+    if ($NetworkMode -eq "l2bridge")
+    {
+        DownloadFile -Url  "https://github.com/$Global:GithubSDNRepository/raw/$Global:GithubSDNBranch/Kubernetes/flannel/l2bridge/cni/win-bridge.exe" -Destination $CniPath\win-bridge.exe
+    }
+    elseif ($NetworkMode -eq "overlay")
+    {
+        DownloadFile -Url  "https://github.com/$Global:GithubSDNRepository/raw/$Global:GithubSDNBranch/Kubernetes/flannel/overlay/cni/win-overlay.exe" -Destination $CniPath\win-overlay.exe
+    }
+}
+
+function DownloadFlannelBinaries()
+{
+    param(
+        [Parameter(Mandatory = $false, Position = 0)]
+        [string] $Release = "0.11.0",
+        [string] $Destination = "c:\flannel"
+    )
+
+    Write-Host "Downloading Flannel binaries"
+    DownloadFile -Url  "https://github.com/coreos/flannel/releases/download/v${Release}/flanneld.exe" -Destination $Destination\flanneld.exe 
+}
+
+function GetKubeFlannelPath()
+{
+    return "c:\etc\kube-flannel"
+}
+
+function InstallFlannelD()
+{
+    param(
+    [Parameter(Mandatory = $false, Position = 0)]
+    [string] $Destination = "c:\flannel",
+    [Parameter(Mandatory = $true)][string] $InterfaceIpAddress
+    )
+    
+    Write-Host "Installing FlannelD Service"
+    $logDir = [io.Path]::Combine($(GetLogDir), "flanneld");
+    CreateDirectory $logDir
+    $log = [io.Path]::Combine($logDir, "flanneldsvc.log");
+
+    DownloadFile -Url  "https://github.com/$Global:GithubSDNRepository/raw/$Global:GithubSDNBranch/Kubernetes/flannel/$Global:NetworkMode/net-conf.json" -Destination $(GetFlannelNetConf)
+    CreateDirectory $(GetKubeFlannelPath)
+    copy $Global:BaseDir\net-conf.json $(GetKubeFlannelPath)
+
+    $flanneldArgs = @(
+        "$Destination\flanneld.exe",
+        "--kubeconfig-file=$(GetKubeConfig)",
+        "--iface=$InterfaceIpAddress",
+        "--ip-masq=1",
+        "--kube-subnet-mgr=1"
+    )
+
+    $service = Get-Service FlannelD -ErrorAction SilentlyContinue
+    if (!$service)
+    {
+        $nodeName = (hostname).ToLower()
+        CreateService -ServiceName FlannelD -CommandLine $flanneldArgs `
+            -LogFile "$log" -EnvVaribles @{NODE_NAME = "$nodeName";}    
+    }
+}
+
+function UnInstallFlannelD()
+{
+    RemoveService -ServiceName FlannelD
+    Remove-Item $(GetKubeFlannelPath) -Force -ErrorAction SilentlyContinue
```

Comment:
I think that's ok because there could be other configs there that weren't created by this script. That directory can be left behind on Linux. For example, you might uninstall the kubelet but have static manifests left in /etc/kubernetes . Then you install a new version and use those files again.

Reply:
It leaves behind an empty ` C:\etc\kubernetes\pki` which seems to follow the kubeadm output of `Deleting contents of config directories: [\etc\kubernetes\manifests /etc/kubernetes/pki]`